File: app/src/main/java/com/sumian/app/network/engine/NetEngine.java
package com.sumian.app.network.engine;

import com.google.gson.GsonBuilder;
import com.sumian.sleepdoctor.BuildConfig;
import com.sumian.app.network.api.SleepyApi;
import com.sumian.app.network.api.SleepyV1Api;
import com.sumian.app.network.interceptor.AuthInterceptor;

import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.TimeUnit;

import okhttp3.OkHttpClient;
import okhttp3.Protocol;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * Created by jzz
 * on 2017/9/26
 * <p>
 * desc:
 */

public class NetEngine {

    private static final int TIMEOUT = 3;
    private SleepyApi mSleepyApi;

    private SleepyV1Api mSleepyV1Api;

    public NetEngine() {

        List<Protocol> protocols = new ArrayList<>();
        protocols.add(Protocol.HTTP_1_1);
        protocols.add(Protocol.HTTP_2);

        OkHttpClient okHttpClient = new OkHttpClient
                .Builder()
                .connectTimeout(TIMEOUT, TimeUnit.SECONDS)
                .readTimeout(TIMEOUT, TimeUnit.SECONDS)
                .writeTimeout(TIMEOUT, TimeUnit.SECONDS)
                .protocols(protocols)
                .addInterceptor(new HttpLoggingInterceptor().setLevel(BuildConfig.DEBUG ? HttpLoggingInterceptor.Level.BODY : HttpLoggingInterceptor.Level.NONE))
                .addInterceptor(AuthInterceptor.create())
                .build();

        Retrofit retrofit = new Retrofit
                .Builder()
                .baseUrl(BuildConfig.BASE_URL)
//            .addConverterFactory(FastJsonConverterFactory.create())
                .addConverterFactory(GsonConverterFactory.create(new GsonBuilder().create()))
                .callFactory(okHttpClient)
                .build();

        this.mSleepyApi = retrofit.create(SleepyApi.class);
        this.mSleepyV1Api = retrofit.create(SleepyV1Api.class);
    }

    public SleepyApi getHttpService() {
        return mSleepyApi;
    }

    public SleepyV1Api getV1HttpService() {
        return mSleepyV1Api;
    }

}

File: app/src/main/java/com/sumian/sleepdoctor/network/engine/NetEngine.kt
package com.sumian.sleepdoctor.network.engine

import com.google.gson.GsonBuilder
import com.sumian.sleepdoctor.BuildConfig
import com.sumian.sleepdoctor.network.api.DoctorApi
import com.sumian.sleepdoctor.network.interceptor.AuthInterceptor
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import java.util.concurrent.TimeUnit

/**
 * Created by jzz
 * on 2018/1/15.
 * desc:
 */

class NetEngine {

    private val mDoctorApi: DoctorApi

    companion object {
        private const val TIMEOUT = 5
    }

    init {

        val okHttpClient = OkHttpClient.Builder()
                .connectTimeout(TIMEOUT.toLong(), TimeUnit.SECONDS)
                .readTimeout(TIMEOUT.toLong(), TimeUnit.SECONDS)
                .writeTimeout(TIMEOUT.toLong(), TimeUnit.SECONDS)
                .addInterceptor(HttpLoggingInterceptor()
                        .setLevel(if (BuildConfig.DEBUG) HttpLoggingInterceptor.Level.BODY else HttpLoggingInterceptor.Level.NONE))
                .addInterceptor(AuthInterceptor.create())
                .build()

        val retrofit = Retrofit.Builder()
                .baseUrl(BuildConfig.BASE_URL)
                .addConverterFactory(GsonConverterFactory.create(GsonBuilder().create()))
                .callFactory(okHttpClient)
                .build()

        this.mDoctorApi = retrofit.create(DoctorApi::class.java)

    }

    fun httpRequest(): DoctorApi {
        return mDoctorApi
    }


}
